### PR TITLE
fix: stop frozen Connect leaking bundled Qt env into launched DCCs

### PR DESCRIPTION
**Pull request by dennisweil**
Resolves : _no dedicated ticket found for this bug — add an `FT-`/`F-` link here if one exists._

- [x] I have added automatic tests where applicable.
- [x] The PR contains a description of what has been changed.
- [x] The description contains manual test instructions.
- [ ] The PR contains update to the release notes.
- [ ] The PR contains update to the documentation.

This PR has been tested on :

- [ ] Windows.
- [x] MacOs.
- [ ] Linux.

# Overview
**Purpose:** Stop a frozen (PyInstaller) ftrack Connect from leaking its own bundled Qt paths into the environment of launched DCCs. Harmony 25.2/27 (Qt 6.4) otherwise loads Connect's bundled Qt 6.6 plugins, printing ~10 `Plugin uses incompatible Qt library (6.6.0)` warnings and breaking menus/dialogs.

**Scope:** `apps/connect` launcher (root fix, all DCCs) and the `framework-harmony` launch hook (transitional fix), plus tests, a CI job, and a design record.

## Implementation Details
**Approach:** Two layers. Layer 1 — the Connect launcher (`_get_application_environment`) strips its own bundle's entries from `QT_PLUGIN_PATH`/`QML(2)_IMPORT_PATH`/`PATH`/loader-library paths before the launch event is published, so any studio-set Qt vars applied afterwards still win; guarded by `sys.frozen` and never touching `os.environ`. Layer 2 — the Harmony hook emits equivalent `.set`/`.unset` launch-env actions so users on an already-installed frozen Connect (predating Layer 1) are protected immediately by the next plugin release.

**Reasoning:** The two layers ship on different release vehicles (Connect vs. the framework-harmony plugin); Layer 2 reaches existing installs now, Layer 1 fixes it at the source for every DCC. A previously-proposed `setLibraryPaths()` patch in the standalone process was rejected (wrong process; runs too late) — see the design record.

**Changes:** New pure helpers `get_frozen_bundle_roots`/`strip_frozen_qt_environment`/`restore_loader_library_path` in the connect launcher; `get_frozen_qt_environment_actions` (self-contained twin, marked `TODO(remove)`) wired into the Harmony hook's `on_launch_integration`.

**Trade-offs:** Layer 2 duplicates Layer 1's logic deliberately (hooks must not depend on new Connect code); the duplicate is idempotent alongside a fixed Connect and is scheduled for removal once the minimum supported Connect ships Layer 1 (> 26.6.x).

## Testing
**Tests Added:** First unit tests in `apps/connect` (`tests/unit/test_application_launcher_environment.py`) covering the frozen-path scrub; a `framework-harmony` `tests/test_qt_env_scrub.py` suite (unit action-emission + live tier-1 launch verification via `ps eww` + tier-2 standalone log check); a `connect-unit-pytest` CI job mirroring `harness-unit-pytest`.

**Manual Testing:** Reproduced on this workstation — launching Harmony 25.2 with `QT_PLUGIN_PATH` = Connect's bundled Qt gave 32 `incompatible Qt library (6.6.0)` lines; with it unset (the scrubbed state) → 0. The Harmony test suite (incl. the live tier-1/tier-2 cases and the existing `test_launch.py`) passed against real Harmony 27. To verify against the frozen Connect: restart `Ftrack Connect.app`, launch Harmony, then `ps eww $(pgrep -f "Harmony 25.2 Premium.app/Contents/MacOS") | grep -oE "QT_PLUGIN_PATH|QML2_IMPORT_PATH"` → expect no output.

**Testing Environment:** macOS (Harmony 25.2 + 27, installed frozen `Ftrack Connect.app`).

## Notes for Reviewers
**Focus Areas:** The env-scrub prefix-matching (`normcase`/`normpath`, sibling-collision guard, macOS two-root `.app` case) and its placement before the launch-event publish so studio overrides still win.

**Dependencies:** None. Layer 1 ships with the next Connect release; Layer 2 with the next framework-harmony plugin release.

**Known Issues:** Release notes and user-facing docs are not yet updated (checkboxes left unchecked). On macOS the DCC's stderr never reaches ftrack's log files, so these warnings were only visible on Windows / via direct stderr capture; the meaningful macOS check is the `ps eww` env inspection above. Windows/Linux end-to-end verification is still pending.

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented bundled (installed/frozen) Connect Qt-related environment settings from leaking into launched applications (including Harmony), covering Qt plugin/QML and related loader-path variables while preserving studio-provided values; restored correct Linux loader library path behavior.
* **Tests**
  * Expanded launcher-side unit coverage plus Harmony hook unit/live integration tests across Windows/macOS/Linux edge cases, and added a dedicated CI run for the Connect unit suite.
* **Documentation**
  * Added a spec detailing the Qt environment leak and the layered remediation approach (launcher scrub + transitional Harmony hook behavior).

### CODE REVIEW

File: projects/framework-harmony/connect-plugin/hook/discover_ftrack_framework_harmony.py [hunk 1]
```diff
@@ -47,6 +47,103 @@ def on_discover_integration(session, event):
...
+# that scrub: this plugin ships in a framework-harmony release that reaches
+# them before the next Connect release does. Remove once the minimum
+# supported Connect version ships the launcher scrub (> 26.6.x).
+#
+# A hook must not depend on new Connect code, so this is a self-contained
+# twin (uses only the already-imported ``os``/``sys``). It is idempotent
+# alongside a fixed Connect: ``.unset`` of an absent key is a no-op and
+# ``.set`` rewrites the identical filtered value. Scope is intentionally
+# the four Qt path vars only (PATH / loader-path handling stays
+# Connect-launcher-only, to keep this transitional surface minimal).
```
[the remark this answers is not in the document]
Comment (dennisweil) [reply projects/framework-harmony/connect-plugin/hook/discover_ftrack_framework_harmony.py]: Intentional scope decision, keeping as-is. This hook is the transitional Layer 2 twin; the design doc deliberately scopes it to the four Qt *plugin-path* vars only and leaves `PATH`/`LD_LIBRARY_PATH`/`DYLD_LIBRARY_PATH` (and the `LD_LIBRARY_PATH_ORIG` restore) to the Connect launcher (Layer 1), to keep this shim minimal and short-lived. Removing the incompatible plugins via `QT_PLUGIN_PATH` stops the reported warnings/breakage; the loader-path leak is fully addressed at the source by Layer 1 for every DCC. Extending the twin to loader paths would widen the transitional surface we intend to delete once Layer 1 is the minimum supported Connect. See docs/specs/2026-07-15-qt-env-leak-frozen-connect-design.md.

File: .github/workflows/push-main-test.yml [hunk 2]
```diff
@@ -69,6 +69,34 @@ jobs:
...
+          python-version: "3.13"
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@d4b2f3b6ecc6e67c4457f6d3e41ec42d3d0fcb86  # v5
+
+      - name: ftrack Connect unit tests
+        working-directory: apps/connect
+        run: |
+          uv sync --extra test
+          uv run pytest tests/
```
[the remark this answers is not in the document]
Comment (dennisweil) [reply .github/workflows/push-main-test.yml:98]: Already handled without provisioning Qt libs: the test loads the launcher module directly from its file (`_load_launcher_module`) rather than importing the `ftrack_connect` package, so `ftrack_connect/__init__.py` (and its `import qtawesome` → PySide6) never runs during collection. No `libegl1` needed on the runner.

File: apps/connect/pyproject.toml [hunk 3]
```diff
@@ -85,6 +85,11 @@ ftrack-constants = { path = "../../libs/constants" }
 ftrack-utils = { path = "../../libs/utils" }
 ftrack-framework-core = { path = "../../libs/framework-core" }
 
+[tool.pytest.ini_options]
+# The launcher module lives under source/; make it importable without an
+# install so `uv run pytest tests/` works from a bare `uv sync --extra test`.
+pythonpath = ["source"]
```
[the remark this answers is not in the document]
Comment (dennisweil) [reply apps/connect/pyproject.toml:91]: Resolved via `apps/connect/conftest.py`, which sets `collect_ignore = ["tests"]` when `ftrack_connect` is not importable. The repo-root `unit-test-pytest` job (no `ftrack_connect` in its venv) therefore skips collecting these tests; the dedicated `connect-unit-pytest` job (which does `uv sync` in `apps/connect`) collects and runs them. Kept the `pythonpath` setting unchanged as suggested.

**Review by hdd — APPROVED** [review]
